modules/fluid_properties/src/fluidproperties/Water97FluidProperties.C

```diff
@@ -446,6 +446,38 @@ Water97FluidProperties::s_from_p_T(
   s_from_p_T_template(pressure, temperature, s, ds_dp, ds_dT);
 }
 
+Real
+Water97FluidProperties::s_from_h_p(Real enthalpy, Real pressure) const
+{
+  Real T = T_from_p_h(pressure, enthalpy);
+  return s_from_p_T(pressure, T);
+}
+
+ADReal
+Water97FluidProperties::s_from_h_p(const ADReal & enthalpy, const ADReal & pressure) const
+{
+  ADReal temperature = T_from_p_h_ad(pressure, enthalpy);
+  return s_from_p_T_template(pressure, temperature);
+}
+
+void
+Water97FluidProperties::s_from_h_p(
+    const Real enthalpy, const Real pressure, Real & s, Real & ds_dh, Real & ds_dp) const
+{
+  ADReal p = pressure;
+  Moose::derivInsert(p.derivatives(), 0, 1.0);
+
+  ADReal h = enthalpy;
+  Moose::derivInsert(h.derivatives(), 1, 1.0);
+
+  ADReal T = T_from_p_h_ad(p, h);
+  ADReal entropy = s_from_p_T_template(p, T);
+
+  ds_dh = entropy.derivatives()[1];
+  ds_dp = entropy.derivatives()[0];
```

Comment:
@aikubo notice I flipped those. if you were getting convergence issues in your model this might have been it